sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -223,11 +224,18 @@ class Dataset[T] private[sql](
   @transient private[sql] val logicalPlan: LogicalPlan = {
     // For various commands (like DDL) and queries with side effects, we force query execution
     // to happen right away to let these side effects take place eagerly.
+    def eagerRun(plan: LogicalPlan): LogicalPlan = {
+      val relation =
+        LocalRelation(plan.output, withAction("command", queryExecution)(_.executeCollect()))
+      relation.setTagValue(Dataset.DATASET_EAGER_RUN_TAG, true)
+      relation
+    }
+
     val plan = queryExecution.analyzed match {
       case c: Command =>
-        LocalRelation(c.output, withAction("command", queryExecution)(_.executeCollect()))
```

Comment:
It's a bit tricky to trigger a SQL execution in the middle of DataFrame execution. I'd prefer the other way around: we don't trigger SQL execution here. We follow the common code path and only trigger SQL execution at the end like `df.collect`.

We can add some additional info to `LocalRelation`, so that its SQL UI shows the original command.

Reply:
But when people run commands without action in Spark application, e.g. `sql("DROP TABLE t")`,  this is the place that Spark implicitly adds a `collect` action to the DataFrame.

Reply:
ah I see, then your idea looks good. One suggestion is, let's not use `TreeNodeTag` here which is unreliable. We can directly add a new `fromCommand` flag in `LocalRelation`

Reply:
Ok.